### ISSUE TITLE
Background chooser dialog: sets default background when it's first opened

### ISSUE THREAD

**Issue by loentar** [mypaint/mypaint#512]
The problem is background always become white when you re-open saved work with non-white background and select "background chooser" for the fist time after loading. "Undo" does not work at this point and it's not possible to restore background without reverting.

How to reproduce:
1. draw something or leave the picture blank
2. open background chooser and set background
3. save your work
4. restart MyPaint -- required to reproduce
5. open saved work
6. ensure background has a color/texture
7. open background chooser -- background become white.

**Comment by achadwick** [comment]
I cannot reproduce this with a fresh config any MyPaint master as of 0428646094568bb2ed5e034e9e86c5a70c7e2228 (= 1.2.0-beta.3) on Debian testing/unstable in GNOME using GTK+ 3.18.2.

* To reliably get a fresh config, I used

  ```sh
  cd /path/to/clone/of/mypaint
  # ... then rebuild according to the instructions, then...
  rm -vfr /tmp/cfgtmp$$ && ./mypaint -c /tmp/cfgtmp$$
  ```

* When I opened the background chooser, I used the Layer menu.
* When I saved, I used "Save As" (via the Ctrl+Shift+S accelerator)
* MyPaint has no restart action, so I closed the program, waited a short while, then started the program again using

  ```sh
  /mypaint -c /tmp/cfgtmp$$
  ```

* When I opened the file previously saved, I used the Ctrl+O accelerator (File → Open...)
* I am using the default Adwaita theme.

In step 7 of my testing, the background stayed exactly as it was when loaded from the background chooser in step 5.

**Comment by achadwick** [comment]
There is not enough information in your report for me to investigate this bug yet. I need to be able to see it for myself really.

* Please expand on your instructions, if you need to say more.
* Please tell me what
  * operating system + version
  * windowing environment + version
  * version of MyPaint **← most important!** -- be as detailed as you can, always use (Help → About MyPaint... and paste the full version number. If you can, please separate out the hex string on the end - it links up to the revision history nicely on Github if you do that!)
  * version of libgtk **← maybe important**
  * GTK theme
  you are using.
* Please tell me how you are invoking the commands.
* Please tell me if this is still happening for you.

Thanks!

**Comment by achadwick** [comment]
Aha! Needs an extra step zero:

0. Set a default background

and then step 2 needs to be modified:

2. open background chooser and set background <ins>to something other than the default</ins>

So in short, opening the background window sets the background to the default, but only the very first time the background window is opened.

**Comment by achadwick** [comment]
Possibly fairly lightweight: loading the default background probably just needs splitting out of the (much slower, and thus deferred) dialog initialization phase.

**Comment by loentar** [comment]
Yes, everything as you said: I've changed default background and It works with clean config.

**Comment by cortexer** [comment]
I was giving a try on this issue.
My solution would be:
Simply don't set the background to the window when loading the background chooser dialog.
Normally the background of the paintings is loaded somewhere else (for example in application.py).
Maybe I am overlooking something and this solution is too simplistic?

I created a pull request on this.
As this is my first approach in doing more than only translations for mypaint, someone should have a look at this.

**Comment by achadwick** [comment]
@cortexer Yeah, that looks like a real legacy of the way things used to be done. Loading the user's default background now happens at application startup, so the bit you've identified needs a bit of cleanup: it's now a source of error. Thanks for the PR (#515): I've left some comments on it!

Moving to the release milestone, since this is being actively looked at! :construction_worker: :construction:

**Comment by achadwick** [comment]
Closed by 0bca464cf50c681ffae57d4459f648ace40cb005 / #515.
